satpy/composites/crefl_utils.py

```diff
@@ -359,44 +459,53 @@ def run_crefl(refl, coeffs,
+        lon[(lon <= -180) | (lon >= 180)] = np.nan
         row = ((90.0 - lat) * avg_elevation.shape[0] / 180.0).astype(np.int32)
         col = ((lon + 180.0) * avg_elevation.shape[1] / 360.0).astype(np.int32)
-        height = avg_elevation[row, col].astype(np.float64)
+        space_mask = da.isnull(lon) | da.isnull(lat)
+        row[space_mask] = 0
+        col[space_mask] = 0
+
+        def _avg_elevation_index(avg_elevation, row, col):
+            return avg_elevation[row, col]
+        height = da.map_blocks(_avg_elevation_index, avg_elevation, row, col, dtype=avg_elevation.dtype)
+        height = xr.DataArray(height, dims=['y', 'x'])
         # negative heights aren't allowed, clip to 0
-        height = height.where(height >= 0., 0.0)
+        height = height.where((height >= 0.) & ~space_mask, 0.0)
         del lat, lon, row, col
-
     mus = xu.cos(xu.deg2rad(solar_zenith))
     muv = xu.cos(xu.deg2rad(sensor_zenith))
     phi = solar_azimuth - sensor_azimuth
 
-    del solar_azimuth, solar_zenith, sensor_zenith, sensor_azimuth
-
-    sphalb, rhoray, TtotraytH2O, tOG = get_atm_variables(
+    if use_abi:
+        LOG.debug("Using ABI CREFL algorithm")
+        a_O3 = [268.45, 0.5, 115.42, -3.2922]
+        a_H2O = [0.0311, 0.1, 92.471, -1.3814]
+        a_O2 = [0.4567, 0.007, 96.4884, -1.6970]
+        G_O3 = G_calc(solar_zenith, a_O3) + G_calc(sensor_zenith, a_O3)
+        G_H2O = G_calc(solar_zenith, a_H2O) + G_calc(sensor_zenith, a_H2O)
+        G_O2 = G_calc(solar_zenith, a_O2) + G_calc(sensor_zenith, a_O2)
+        # Note: bh2o values are actually ao2 values for abi
+        sphalb, rhoray, TtotraytH2O, tOG = get_atm_variables_abi(
+            mus, muv, (phi + 180.0), height, (ah2o, bh2o, ao3, tau), G_O3, G_H2O, G_O2)
+    else:
+        LOG.debug("Using original VIIRS CREFL algorithm")
+        sphalb, rhoray, TtotraytH2O, tOG = get_atm_variables(
         mus, muv, phi, height, (ah2o, bh2o, ao3, tau))
 
+    del solar_azimuth, solar_zenith, sensor_zenith, sensor_azimuth
+
     if rhoray.shape[1] != refl.shape[1]:
-        LOG.debug(
-            "Interpolating CREFL calculations for higher resolution bands")
-        # Assume we need to interpolate
-        # FIXME: Do real bilinear interpolation instead of "nearest"
-        factor = int(refl.shape[1] / rhoray.shape[1])
-        rhoray = np.repeat(np.repeat(rhoray, factor, axis=0), factor, axis=1)
-        tOG = np.repeat(np.repeat(tOG, factor, axis=0), factor, axis=1)
-        TtotraytH2O = np.repeat(
-            np.repeat(TtotraytH2O, factor, axis=0),
-            factor, axis=1)
-        # if average height wasn't provided then this should stay a scalar
-        if sphalb.size != 1:
-            # otherwise make it the same size as the other arrays
-            sphalb = np.repeat(np.repeat(sphalb, factor, axis=0), factor, axis=1)
+        from satpy.composites import IncompatibleAreas
+        raise IncompatibleAreas("")
```

Comment:
I would say remove this entire if block. This should theoretically not be possible when called from the modifier (because of `check_areas`).